src/test/basic.test.ts

```diff
@@ -910,7 +910,7 @@ test(
 
     // Run both from the workspaces root package.
     {
-      const exec = rig.exec('pnpm run cmd --recursive');
```

Comment:
Re-iterating what you wrote in the description. This broke because `--recursive` ended up passed in as an argv as it came after the script name in pnpm 7. Re-ordering removes `--recursive` from the commands argv.

Reply:
Yep!